src/fs_io.h

```diff
@@ -30,16 +30,17 @@ class dread_ctrl
     void *client_data;
 };
 
-// POD
 class dwrite_q
 {
+    MEMPROXY_CLASS(dwrite_q);
+
 public:
-    off_t file_offset;
-    char *buf;
-    size_t len;
-    size_t buf_offset;
-    dwrite_q *next;
-    FREE *free_func;
+    off_t file_offset = 0;
+    char *buf = nullptr;
+    size_t len = 0;
+    size_t buf_offset = 0;
+    dwrite_q *next = nullptr;
+    FREE *free_func = {}; // function to free 'buf' (if any)
```

Comment:
Please use Doxygen comments when describing data members. If possible, please use nullptr to initialize pointer members.

It is easy to misinterpret "to free buf (if any)" as "to free buf if buf is not nil". In reality, the function is called even for nil buffers. Let's clarify:

```suggestion

private:
    /// when set, gets called upon object destruction to free buf
    FREE *free_func = nullptr;
```